redwood-lazylayout-widget/src/commonMain/kotlin/app/cash/redwood/lazylayout/widget/LazyListUpdateProcessor.kt

```diff
@@ -302,7 +302,7 @@ public abstract class LazyListUpdateProcessor<V : Any, W : Any> {
   private fun maybeShiftLoadedWindow(splitIndex: Int) {
     if (loadedItems.size != 0) return
 
-    if (splitIndex < itemsBefore.size) {
+    if (splitIndex < itemsBefore.size && itemsAfter.size > 0) {
```

Comment:
Unfortunately this is a host-side fix. It might be the right fix for our bug but it won’t apply immediately.

Reply:
Ah right, good point